micrometer-core/src/main/java/io/micrometer/core/instrument/internal/TimedScheduledExecutorService.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright 2019 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.internal;
+
+import io.micrometer.core.instrument.*;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.*;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * A {@link ScheduledExecutorService} that is timed
+ *
+ * @author Sebastian Lövdahl
+ */
+public class TimedScheduledExecutorService implements ScheduledExecutorService {
+    private final MeterRegistry registry;
+    private final ScheduledExecutorService delegate;
+    private final Timer executionTimer;
+    private final Timer idleTimer;
+    private final Counter scheduledOnce;
+    private final Counter scheduledRepetitively;
+
+    public TimedScheduledExecutorService(MeterRegistry registry, ScheduledExecutorService delegate, String executorServiceName, Iterable<Tag> tags) {
+        this.registry = registry;
+        this.delegate = delegate;
+        // TODO: IMO, it would make more sense to name this timer "executor.execution", but that's a breaking change
```

Comment:
This is discussed in #1080 as we're not consistent on the naming. However, we do not want to break existing users. In Micrometer 2.0, we can correct this. In the meantime, we either need to live with the inconsistency and not as specific name, or we need to duplicate the timer so there's both `executor` and `executor.execution` until 2.0 when we can remove `executor`.